apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloAutoPersistedQueryInterceptor.java

```diff
@@ -68,13 +80,22 @@ public void interceptAsync(@NotNull final InterceptorRequest request, @NotNull f
           if (isPersistedQueryNotFound(response.errors())) {
             logger.w("GraphQL server couldn't find Automatic Persisted Query for operation name: "
                 + request.operation.name().name() + " id: " + request.operation.operationId());
-            return Optional.of(request.toBuilder().sendQueryDocument(true).build());
+
+            return Optional.of(request.toBuilder()
+                    .sendQueryDocument(true)
+                    .autoPersistQueries(true)
+                    .useHttpGetMethodForQueries(useHttpGetMethodForQueries)
+                    .build());
           }
 
           if (isPersistedQueryNotSupported(response.errors())) {
             // TODO how to disable Automatic Persisted Queries in future and how to notify user about this
             logger.e("GraphQL server doesn't support Automatic Persisted Queries");
-            return Optional.of(request.toBuilder().sendQueryDocument(true).build());
+            return Optional.of(request.toBuilder()
+                    .sendQueryDocument(true)
+                    .autoPersistQueries(true)
+                    .useHttpGetMethodForQueries(useHttpGetMethodForQueries)
```

Comment:
I believe we don't need this as `request ` is our original request (not modified). So it will have already set this flag.

See line with:
```
chain.proceedAsync(newRequest, dispatcher, new CallBack() {
      @Override public void onResponse(@NotNull InterceptorResponse response) {
        if (disposed) return;
        Optional<InterceptorRequest> retryRequest = handleProtocolNegotiation(request, response);
```

`request` is original request not modified.

Reply:
I see, we can access to original request.

However, we still need to send the httpbody with `extension.persistedQuery` for creating PQ cache in server for later request. So we need `.sendQueryDocument(true) and .autoPersistQueries(true)` and  can't guarantee original request those value on.

to combine the above two comment, probably modified as following? 

```
return Optional.of(request.toBuilder()
                    .sendQueryDocument(true)
                    .autoPersistQueries(true)
                    .useHttpGetMethodForQueries(request.useHttpGetMethodForQueries)

```

Reply:
yes you r right, as original request is preserved, can reuse that one. Fixed as suggestion above.

Beware that we still need to send the http body with `extension.persistedQuery` for creating PQ cache in server for later request. So we need `.sendQueryDocument(true) and .autoPersistQueries(true)` and guarantee original request has those value on.

And it's value depends in RealApolloCall.java line. 139
```
    ApolloInterceptor.InterceptorRequest request = ApolloInterceptor.InterceptorRequest.builder(operation)
        .cacheHeaders(cacheHeaders)
        .requestHeaders(requestHeaders)
        .fetchFromCache(false)
        .optimisticUpdates(optimisticUpdates)
        .useHttpGetMethodForQueries(useHttpGetMethodForQueries)
        .autoPersistQueries(enableAutoPersistedQueries)
        .build();
    interceptorChain.proceedAsync(request, dispatcher, interceptorCallbackProxy());
  }
```